ddl/column_type_change_test.go

```diff
@@ -288,3 +288,158 @@ func init() {
 	mockTerrorMap[model.StateWriteOnly.String()] = terror.ClassDDL.New(1, "MockRollingBackInCallBack-"+model.StateWriteOnly.String())
 	mockTerrorMap[model.StateWriteReorganization.String()] = terror.ClassDDL.New(1, "MockRollingBackInCallBack-"+model.StateWriteReorganization.String())
 }
+
+func (s *testColumnTypeChangeSuite) TestColumnTypeChangeFromIntegerToOthers(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec("use test")
+	// Enable column change variable.
+	tk.Se.GetSessionVars().EnableChangeColumnType = true
+	defer func() {
+		tk.Se.GetSessionVars().EnableChangeColumnType = false
+	}()
+
+	prepare := func(tk *testkit.TestKit) {
+		tk.MustExec("drop table if exists t")
+		tk.MustExec("create table t(a tinyint, b smallint, c mediumint, d int, e bigint, f bigint)")
+		tk.MustExec("insert into t values(1, 11, 111, 1111, 11111, 111111)")
+	}
+	prepareForEnumSet := func(tk *testkit.TestKit) {
+		tk.MustExec("drop table if exists t")
+		tk.MustExec("create table t(a tinyint, b smallint, c mediumint, d int, e bigint)")
+		tk.MustExec("insert into t values(1, 1, 1, 11111, 11111)")
+	}
+
+	// integer to string
+	prepare(tk)
+	tk.MustExec("alter table t modify a varchar(10)")
+	modifiedColumn := getModifyColumn(c, tk.Se.(sessionctx.Context), "test", "t", "a", false)
```

Comment:
I think type-assertion is not necessary.

Reply:
both string/integer type can read the result as "1", so maybe the type check is ok